esmvalcore/preprocessor/_other.py

```diff
@@ -38,3 +39,97 @@ def clip(cube, minimum=None, maximum=None):
             raise ValueError("Maximum should be equal or larger than minimum.")
     cube.data = da.clip(cube.core_data(), minimum, maximum)
     return cube
+
+
+def add_cell_measure(cube, fx_variables, project, dataset, check_level):
+    """
+    Load requested fx files, check with CMOR standards and add the
+    fx variables as cell measures in the cube containing the data.
+
+    Parameters
+    ----------
+    cube: iris.cube.Cube
+        Iris cube with input data.
+    fx_variables: dict
+        Path to the needed fx_files.
+    project: str
+
+    dataset: str
+
+    check_level: CheckLevels
+        Level of strictness of the checks.
+
+
+    Returns
+    -------
+    iris.cube.Cube
+        Cube with added cell measures.
+    """
+    from esmvalcore.preprocessor._io import concatenate
+    from esmvalcore.cmor.fix import fix_metadata, fix_data
+    from esmvalcore.cmor.check import cmor_check_metadata, cmor_check_data
+
+    if not fx_variables:
+        return cube
+    fx_cubes = iris.cube.CubeList()
+    for fx_files in fx_variables.values():
+        if isinstance(fx_files, str):
+            fx_files = [fx_files]
+        if not fx_files:
+            continue
+        for fx_file in fx_files:
+            loaded_cube = iris.load(fx_file)
+            short_name = loaded_cube[0].var_name
+            mip = loaded_cube[0].attributes['table_id']
+            freq = loaded_cube[0].attributes['frequency']
```

Comment:
It might be better to pass these in as function arguments rather than relying on the input NetCDF files to have these global attributes